receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -84,15 +85,16 @@ func (s *scraper) scrape(_ context.Context) (pmetric.Metrics, error) {
 		}
 		errors.AddPartial(0, fmt.Errorf("failed collecting partitions information: %w", err))
 	}
+	translatedPartitions := translateMountsRootPath(partitions, s.config.RootPath)
```

Comment:
The way it currently works, it need both the translated and untranslated mountpoints. The current flow is: 

1. Get the mountpoints from `${root_path}/proc/1/mountinfo`
2. Those mountpoints, because they are delivered from the host, don't contain the `root_path` prefix, so that needs to be added, otherwise we would get a file not found error. Thus we get `translatedPartitions` which is prefixed with `root_path`.
3. When we want to record the metric, we want the `mountpoint` attribute to be from the host's perspective, not the container's perspective. So we use the original, untranslated mountpoint for that. 

There is the option of doing it the other way around too, i.e. get the mountpoints from the container proc and then remove the prefix when recording the metric. I would be interested to hear people's thoughts.

Reply:
I should also point out that if `root_path` is set, the user needs to specifies which filesystems to query from the host's perspective. This is because we expect the reported `mountpoint` attribute to be reported from the host's perspective. 

It's a little bit tricky, and there are a few ways it could be done. This is ultimately what I came to though.

Reply:
What you could do here instead of copying the mounts, is:

```golang
for i, parition := range parition {
  // Filtering here

  // Comment here about why the filepath.Join is needed here.
  usage, err := s.usage(filepath.Join(s.rootfs, partition.Mountpoint))
}
```
I suspect the above my not work for windows, but if the rootfs value is the zerovalue, the string should have no changes?

Reply:
Talked with @MovieStoreGuy and implemented something similar to what he has here. Makes it cleaner!